wgpu-core/src/device/mod.rs

```diff
@@ -677,13 +694,60 @@ impl<A: HalApi> Device<A> {
             return Err(resource::CreateTextureError::InvalidMipLevelCount(mips));
         }
 
-        let raw = unsafe {
+        let raw_texture = unsafe {
             self.raw
                 .create_texture(&hal_desc)
                 .map_err(DeviceError::from)?
         };
 
-        let mut texture = self.create_texture_from_hal(raw, self_id, desc, format_features);
+        let clear_mode = if hal_usage.contains(hal::TextureUses::DEPTH_STENCIL_WRITE)
+            || hal_usage.contains(hal::TextureUses::COLOR_TARGET)
+        {
+            let usage = if desc.format.describe().sample_type == wgt::TextureSampleType::Depth {
+                hal::TextureUses::DEPTH_STENCIL_WRITE
+            } else {
+                hal::TextureUses::COLOR_TARGET
+            };
+            let dimension = match desc.dimension {
+                wgt::TextureDimension::D1 => wgt::TextureViewDimension::D1,
+                wgt::TextureDimension::D2 => wgt::TextureViewDimension::D2,
+                wgt::TextureDimension::D3 => wgt::TextureViewDimension::D2, // Todo? Is this even specified?
```

Comment:
I vaguely remember that not all Metal allows this. So WebGPU baseline probably forbids creation of such things at the API level. That doesn't mean we can't do it internally, but Metal may require a bit more specific code here. Let's defer to follow-ups and talk about this separately.

Ah, also, I think we are mostly agreeing with the group to make 3D textures non-renderable. So there is that :)